### ISSUE TITLE
Enabling gradient checkpointing for VAE in Stable Diffusion Pipeline/Model

### ISSUE THREAD

**Issue by iamsimha** [huggingface/diffusers#1883]
**Is your feature request related to a problem? Please describe.**

For an experiment, I am interested in finetuning VAE present in stable diffusion pipeline. I dont have access to a GPU with large VRAM. Enabling gradient checkpointing for VAE model will help me finetune the model even with low GPU memory. 

**Describe the solution you'd like**
Similar to the UNET model https://github.com/huggingface/diffusers/blob/21bbc633c4d7b9bb7f74caf4b248c6a4079a85c6/src/diffusers/models/unet_2d_condition.py#L346

Implement gradient checkpointing for various modules in VAE.

DISCLAIMER: I am not familiar enough with the code base to know if there are any reasons which makes similar implementation not feasible.

**Describe alternatives you've considered**
In addition to gradient checkpointing, methods like 8 bit adam, float 16 could be used. However, these reduced precision methods donot calculate the gradient accurately. Having gradient checkpointing will help reduce memory costs significantly, without compromising on the gradient precision.

**Comment by patrickvonplaten** [comment]
Hey @iamsimha,

Currently I'm sadly don't have the time to add gradient checkpointing to the VAE, but it would definitely make sense! 

Usually people don't fine-tune VAE which is also my it'll be difficult to prioritize this issue.

If someone from the community is interested in taking this one this would be great.

**Comment by iamsimha** [comment]
Thanks @patrickvonplaten for the reply. I can give a shot at adding gradient checkpointing for VAE.

**Comment by Pie31415** [comment]
Is there a PR up yet for this issue? If not, I'll take a crack at it.

**Comment by williamberman** [comment]
thank you @Pie31415 !
